server/mdm/apple/gdmf/api_test.go

```diff
@@ -0,0 +1,250 @@
+package gdmf
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	apple_mdm "github.com/fleetdm/fleet/v4/server/mdm/apple"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetLatest(t *testing.T) {
+	// test GetLatestOSVersion using a mock server that returns a known response
+	// and ensure the response is parsed correctly
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		// load the test data from the file
+		b, err := os.ReadFile("test_data.json")
+		require.NoError(t, err)
+		_, err = w.Write(b)
+		require.NoError(t, err)
+	}))
+	os.Setenv("FLEET_DEV_GDMF_URL", srv.URL)
+	t.Cleanup(func() {
+		srv.Close()
+		os.Unsetenv("FLEET_DEV_GDMF_URL")
+	})
```

Comment:
Just a heads-up, there's now a `t.Setenv` method to replace that sequence of calls: https://pkg.go.dev/testing#T.Setenv